it/src/main/java/com/google/cloud/teleport/it/bigtable/DefaultBigtableResourceManager.java

```diff
@@ -0,0 +1,390 @@
+import static com.google.cloud.teleport.it.bigtable.BigtableResourceManagerUtils.generateDefaultClusters;
+import static com.google.cloud.teleport.it.bigtable.BigtableResourceManagerUtils.generateInstanceId;
+import static com.google.cloud.teleport.it.common.ResourceManagerUtils.checkValidProjectId;
+
+import com.google.api.gax.core.CredentialsProvider;
+import com.google.api.gax.rpc.NotFoundException;
+import com.google.api.gax.rpc.ServerStream;
+import com.google.cloud.bigtable.admin.v2.BigtableInstanceAdminSettings;
+import com.google.cloud.bigtable.admin.v2.BigtableTableAdminSettings;
+import com.google.cloud.bigtable.admin.v2.models.CreateInstanceRequest;
+import com.google.cloud.bigtable.admin.v2.models.CreateTableRequest;
+import com.google.cloud.bigtable.admin.v2.models.StorageType;
+import com.google.cloud.bigtable.data.v2.BigtableDataClient;
+import com.google.cloud.bigtable.data.v2.BigtableDataSettings;
+import com.google.cloud.bigtable.data.v2.models.Query;
+import com.google.cloud.bigtable.data.v2.models.Row;
+import com.google.cloud.bigtable.data.v2.models.RowMutation;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterables;
+import java.io.IOException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Default class for implementation of {@link BigtableResourceManager} interface.
+ *
+ * <p>The class supports one instance, and multiple tables per manager object. An instance is
+ * created when the first table is created if one has not been created already.
+ *
+ * <p>The instance id is formed using testId. The instance id will be "{testId}-{ISO8601 time,
+ * microsecond precision}", with additional formatting. Note: If testId is more than 30 characters,
+ * a new testId will be formed for naming: {first 21 chars of long testId} + “-” + {8 char hash of
+ * testId}.
+ *
+ * <p>The class is thread-safe.
+ */
+public class DefaultBigtableResourceManager implements BigtableResourceManager {
+
+  private static final Logger LOG = LoggerFactory.getLogger(DefaultBigtableResourceManager.class);
+  private static final int MAX_TEST_ID_LENGTH = 30;
+  private static final String DEFAULT_CLUSTER_ZONE = "us-central1-a";
+  private static final int DEFAULT_CLUSTER_NUM_NODES = 1;
+  private static final StorageType DEFAULT_CLUSTER_STORAGE_TYPE = StorageType.SSD;
+
+  private final String projectId;
+  private final String instanceId;
+  private final String testId;
+  private final BigtableResourceManagerClientHandler bigtableResourceManagerClientHandler;
+
+  private boolean hasInstance = false;
+
+  private DefaultBigtableResourceManager(DefaultBigtableResourceManager.Builder builder)
+      throws IOException {
+    // Check that the project ID conforms to GCP standards
+    checkValidProjectId(builder.projectId);
+
+    this.projectId = builder.projectId;
+    this.instanceId = builder.instanceId;
+    this.testId = builder.testId;
```

Comment:
`testId` seems to be unused.
There is a test that checks its values, but why do we need to check the value of a field that's not used anywhere else? I would remove both this field and the test case.

Reply:
Fixed.